pkg/cmd/roachtest/tests/mixed_version_cdc.go

```diff
@@ -561,3 +586,67 @@ func runCDCMixedVersions(ctx context.Context, t test.Test, c cluster.Cluster) {
 	mvt.AfterUpgradeFinalized("wait and validate", tester.waitAndValidate)
 	mvt.Run()
 }
+
+// runCDCMixedVersionCheckpointing tests that the new span-level checkpoint
+// code added in 25.2 works correctly in mixed-version states.
+//
+// Writing the checkpoint is explicitly forced with cluster settings and
+// restoring the checkpoint implicitly happens following each of the rolling
+// restarts during the mixed-version test run.
+func runCDCMixedVersionCheckpointing(ctx context.Context, t test.Test, c cluster.Cluster) {
+	tester := newCDCMixedVersionTester(ctx, c)
+
+	mvt := mixedversion.NewTest(
+		ctx, t, t.L(), c, tester.crdbNodes,
+		// We're only concerned with mixed-version compatibility starting at
+		// versions that can upgrade to 25.2 (only 24.3 and 25.1), since that's
+		// the first version with the new span-level checkpoint format.
+		mixedversion.MinimumBootstrapVersion("v24.3.0"),
+	)
+
+	cleanupKafka := tester.StartKafka(t, c)
+	defer cleanupKafka()
+
+	forceCheckpointing := func(
+		ctx context.Context, l *logger.Logger, r *rand.Rand, h *mixedversion.Helper,
+	) error {
+		// NB: We use the 24.3 names for the cluster settings so that the cluster
+		// can understand them at startup time.
+		for _, stmt := range []string{
+			`SET CLUSTER SETTING changefeed.frontier_checkpoint_frequency = '1s'`,
+			`SET CLUSTER SETTING changefeed.frontier_highwater_lag_checkpoint_threshold = '1us'`,
+		} {
+			if err := h.System.Exec(r, stmt); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	scatter := func(
+		ctx context.Context, l *logger.Logger, r *rand.Rand, h *mixedversion.Helper,
+	) error {
+		return h.Exec(r, `ALTER TABLE bank.bank SCATTER`)
+	}
+
+	// Test setup.
+	mvt.OnStartup("start changefeed", tester.createChangeFeed)
+	mvt.OnStartup("create validator", tester.setupValidator)
+	mvt.OnStartup("init workload", tester.initWorkload)
+	mvt.OnStartup("set checkpointing settings", forceCheckpointing)
+
+	// Run workload and kafka consumer.
+	runWorkloadCmd := tester.runWorkloadCmd(mvt.RNG())
+	_ = mvt.BackgroundCommand("run workload", tester.workloadNodes, runWorkloadCmd)
```

Comment:
super nit: mixed version framework already prepends "run" in front of all the hooks so this would be printed as `run run workload`

You might have to run the test 100 more times after this change 😞

Reply:
Maybe the framework shouldn't prepend "run" in front 😜

I'm going to keep it as is because I like having a verb for the action